src/treq/multipart.py

```diff
@@ -267,7 +263,7 @@ def _converted(fields):
             filename = _enforce_unicode(filename) if filename else None
             yield name, (filename, content_type, producer)
 
-        elif isinstance(value, (bytes, text_type)):
+        elif isinstance(value, (bytes, str)):
             yield name, _enforce_unicode(value)
```

Comment:
This seems suspicious (to check for str then call a thing called `_enforce_unicode`) -- I didn't go look what that function does though.

Reply:
if you have `str` `_enforce_unicode` just return the value

https://github.com/twisted/treq/pull/320/files#diff-3bfa11ae9c777a650bf927c1e7d7b062a73451b16151d06466a75405c8d4c28eL228


----

I  am expecting that in real life, you will only have `bytes` here...and never `str`

It would be interesting to split this into 2 separate conditions and see if we have test coverage for both branches.

Reply:
`_enforce_unicode` decodes `bytes` to `str` using UTF-8. It raises a `TypeError` for non-string values. Given that, this does not appear incorrect.